### PR TITLE
Refactor FileBrowser's filterItems function

### PR DESCRIPTION
**Pull request by Wallacoloo**
FileBrowser used to have two `filterItems` functions that shared 90% of the same code (one for filtering the root items and one for recursively filtering their children).

This PR merges these two functions to decrease the amount of duplicate code. Additionally, I made the function private instead of public and I fixed some formatting throughout the file. Lastly, I changed the context menu entries when right-clicking on presets to use "Pattern Editor" instead of "B+B Editor", to be consistent with the rest of the program.

### CODE REVIEW

**Comment by curlymorphic** [comment]
Nice work tidying up this area of code. I have read through the changes, and yet again I can find nothing to comment on :)

**Comment by Wallacoloo** [comment]
Thanks for looking it over @curlymorphic :)

File: src/gui/FileBrowser.cpp [hunk 1]
```diff
@@ -403,11 +340,11 @@ void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent * e )
 						this,
 					SLOT( sendToActiveInstrumentTrack() ) );
 		contextMenu.addAction( tr( "Open in new instrument-track/"
-								"Song-Editor" ),
+								"Song Editor" ),
 						this,
 					SLOT( openInNewInstrumentTrackSE() ) );
 		contextMenu.addAction( tr( "Open in new instrument-track/"
-								"B+B Editor" ),
+								"Pattern Editor" ),
```
Comment (tresf) [inline src/gui/FileBrowser.cpp]: @Wallacoloo I'm not sure this got caught, but this cause inconsistencies in the interface... 

image

Comment (Wallacoloo) [inline src/gui/FileBrowser.cpp]: @tresf I don't see the inconsistency. Can you clarify what the issue is?

Comment (tresf) [inline src/gui/FileBrowser.cpp]: We haven't renamed "Beat/Bassline Editor" to "Pattern Editor" yet.

Comment (Wallacoloo) [inline src/gui/FileBrowser.cpp]: @tresf Oh. That's awkward. I saw `View -> Pattern Editor` had been updated, so I assumed these were leftover areas that had been missed. But now I see many/most places still use B&B editor (e.g. right-clicking on a bassline pattern, and also the default B&B pattern names & the window title for the B&B editor).

Was the change in the view menu a rogue edit? I can submit rollback patches for that plus the change I made, if desired.

Comment (tresf) [inline src/gui/FileBrowser.cpp]: > Was the change in the view menu a rogue edit? I can submit rollback patches for that plus the change I made, if desired.

I'm not sure, but the vote seems to be to move ahead with "Pattern Editor" long term, so a separate fixup commit would be just fine until that happens (and its probably best to leave the name change to occur when the class rename occurs too).

**Pull request merged**